### PR TITLE
read|write_dev : fix bug & qualify channel names and err if bad name

### PR DESCRIPTION
**Pull request by rgetz**
Two small commits:

1. Fix a bug that Paul found that I introduced. in iio_read|write_dev - which was an oversight in a previous commit. allow multiple channels to be added on the command line.
2. In the past, you could put gibberish channel names, and read|write_dev would accept them (ignoring them). This sometimes lead to typos causing issues with "wrong" or "incorrect" data. Fix this by checking for a match, and errorring if there is something provided that doesn't match.
    
Signed-off-by: Robin Getz <robin.getz@analog.com>

### CODE REVIEW

**Review by pcercuei — CHANGES_REQUESTED** [review]

File: tests/iio_readdev.c [hunk 1]
```diff
@@ -373,16 +378,40 @@ int main(int argc, char **argv)
...
+						iio_context_destroy(ctx);
+						return EXIT_FAILURE;
+					} else {
+						iio_channel_enable(ch);
+						nb_active_channels++;
+						ret = 0;
+						break;
+					}
+				}
+			}
```
[the remark this answers is not in the document]
Comment (rgetz) [reply tests/iio_readdev.c]: >I think you can do much better here, 

Thanks for feedback, appreciate it. I always tell people "aim higher". :)

updated with your ideas (mostly).

File: tests/iio_common.c [hunk 2]
```diff
@@ -126,6 +126,36 @@ struct iio_context * autodetect_context(bool rtn, const char * name, const char
 	return ctx;
 }
 
+/* How many input or output channels are in the device */
```
Comment (pcercuei) [inline tests/iio_common.c:129]: That doesn't really seem to describe what this function does.

Comment (rgetz) [reply tests/iio_common.c]: yeah, sorry about that. dropped.

File: tests/iio_readdev.c [hunk 3]
```diff
@@ -373,16 +378,40 @@ int main(int argc, char **argv)
...
+					struct iio_channel *ch = iio_device_get_channel(dev, i);
+					if (iio_channel_is_scan_element(ch) &&
+							!iio_channel_is_output(ch)) {
+						const char *n = iio_channel_get_name(ch);
+						const char *id = iio_channel_get_id(ch);
+						if (n)
+							fprintf(stderr, " %s", n);
+						if (id)
+							fprintf(stderr, " %s", id);
+					}
```
[the remark this answers is not in the document]
Comment (rgetz) [reply tests/iio_readdev.c]: removed

File: tests/iio_readdev.c [hunk 4]
```diff
@@ -322,6 +322,11 @@ int main(int argc, char **argv)
 		iio_context_destroy(ctx);
 		return EXIT_FAILURE;
 	}
+	if (argc > optind + (int)iio_device_get_scan_channels_count(dev, false) + 1) {
+		fprintf(stderr, MY_NAME " : Too many channels specified\n");
+		iio_context_destroy(ctx);
+		return EXIT_FAILURE;
+	}
```
[the remark this answers is not in the document]
Comment (rgetz) [reply tests/iio_readdev.c]: removed

**Review by pcercuei — APPROVED** [review]

File: tests/iio_common.c [hunk 5]
```diff
@@ -126,6 +126,23 @@ struct iio_context * autodetect_context(bool rtn, const char * name, const char
 	return ctx;
 }
 
+ssize_t iio_device_enable_channel(const struct iio_device *dev, const char * channel, bool type)
+{
+	struct iio_channel *ch;
+	if (!channel)
+		return -EFAULT;
```
Comment (pcercuei) [inline tests/iio_common.c:133]: I don't think you need to check that. This will never happen.

File: tests/iio_common.c [hunk 6]
```diff
@@ -126,6 +126,23 @@ struct iio_context * autodetect_context(bool rtn, const char * name, const char
 	return ctx;
 }
 
+ssize_t iio_device_enable_channel(const struct iio_device *dev, const char * channel, bool type)
```
Comment (pcercuei) [inline tests/iio_common.c:129]: Function return type could be just `int`.

Comment (rgetz) [reply tests/iio_common.c]: I think that caused an issue/warning on Windows - will double check...

File: tests/iio_common.c (hunk 5 above)
Comment (rgetz) [reply tests/iio_common.c]: today - but not sure about how others might use us :) 

Removed.

File: tests/iio_common.c (hunk 6 above)
Comment (rgetz) [reply tests/iio_common.c]: yeah, I didn't see the same warnings this time - changed and squashed.

**Comment by pcercuei** [comment]
Merged, thanks!

**Pull request merged**